initial commit (#1)
logzio/synthetic-scripting: README.md

# synthetic-scripting

> 

Reviewer: Before I start reviewing the code, its hard to tell what this integration is used for. Can you add a short description to the readme of what this integration actually does?

## Run Application

> 1. Pull the Docker image

Reviewer 2: ```suggestion
1. Pull the Docker image.
```

`docker pull logzio/synthetic-script:latest`

> 2. Run the docker

Reviewer 2: ```suggestion
2. Run the Docker container.
```

`docker run -p 8080:8080 -d --name synthetic-script logzio/synthetic-script:latest`

> Docker image will run Node js mini app. Go to http://localhost:8080, on you Browser you will see UI that help you to create Connection between your app and logz.io.

Reviewer 2: ```suggestion
Docker image will run a Node.js mini app. Open http://localhost:8080 in your Browser to access the UI that helps you establish connection between your app and Logz.io.
```

> ## Define script for running test

Reviewer 2: ```suggestion
## Define a script for running the test
```

> 1. On a First tab `General Settings` you will see Please fill all reqiured fields.

Reviewer 2: ```suggestion
1. In tab `General Settings`, fill in all the required fields.
```

> ![UI first screen](assets/first-screen.png)

Reviewer: Should the user press accept configs before testing locally? it is not clear

| Field                                                    | Description                                                                                                               |
| -------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- |

> | Name of the Lambda (Required)                            | Lambda function name what will be created and it also will be a name of the text to represent it inside logz.io Dashboard |

Reviewer 2: ```suggestion
| Name of the Lambda (Required)                            | Lambda function name what will be created. This name will also be used to identify the Lambda function in Logz.io Dashboard |
```

> | Description (Optional)                                   | Lambda function description what will be create                                                                           |

Reviewer 2: ```suggestion
| Description (Optional)                                   | Lambda function description                                                                           |
```

> | Logz.io shipping Token (Required)                        | Your Logz.io metrics account token.                                                                                       |

Reviewer 2: ```suggestion
| Logz.io shipping Token (Required)                        | Your Logz.io logs shipping token.                                                                                       |
```

> | Range time for send data to the logs (Default: 1 minute) | Define a time in a minutes to run a lambda function (using cloudBridge event)                                             |

Reviewer 2: ```suggestion
| Time range for sending logs (Default: 1 minute) | Define the range in a minutes to run a Lambda function (using cloudBridge event)                                             |
```

| ------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Aws Access Key (Required)             | Your AWS access key ID.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |

> | Aws Secret Key (Required)             | Your AWS secret key ID.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |

Reviewer: We need to tell our users what are the minimum permissions required for the iam user, for this integration to work

> | Bucket Name (Required)                | The bucket name to which the PUT action was initiated. When using this action with an access point, you must direct requests to the access point hostname. The access point hostname takes the form AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com. When using this action with an access point through the Amazon Web Services SDKs, you provide the access point ARN in place of the bucket name. For more information about access point ARNs, see Using access points in the Amazon S3 User Guide. |

Reviewer 2: ```suggestion
| Bucket Name (Required)                | The bucket name to which the PUT action is initiated. When using this action with an access point, you must direct requests to the access point hostname. The access point hostname takes the form AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com. When using this action with an access point through the Amazon Web Services SDKs, you need to provide the access point ARN instead of the bucket name. For more information about access point ARNs, see Using access points in the Amazon S3 User Guide. |
```

> | Aws Region (Required)                 | Your region to send service requests to.                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer 2: ```suggestion
| Aws Region (Required)                 | Your AWS region to send service requests to.                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

| Aws AccountId (Required)              | AWS account ID.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| AWS IAM Role Lambda Create (Required) | The Amazon Resource Name (ARN) of the function's execution role.                                                                                                                                                                                                                                                                                                                                                                                                                                                      |

> | AWS IAM Role Event Put (Required)     | Add Rule for put event to lambda function                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |

Reviewer 2: ```suggestion
| AWS IAM Role Event Put (Required)     | Add a rule for PUT events to the Lambda function.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```

2. On a Third tab `Enviroment Variables` you will see.

![UI third screen](assets/third-screen.png)

> You can define here env variable that can be attach for your lambda function and you can use it in Code Editor as `process.env.KEY` where Key you can define in that section and it will be prepresent inside you lambda function

Reviewer 2: ```suggestion
Here you can define the environment variable that can be attached to your Lambda function. You can use it in the Code Editor as `process.env.KEY`. You can define the KEY parameter, which will be present in the Lambda function
```

> You can define here env variable that can be attach for your lambda function and you can use it in Code Editor as `process.env.KEY` where Key you can define in that section and it will be prepresent inside you lambda function

Reviewer 2: What will the KEY parameter do exactly?

3. On A Fourth tab `Code Editor` you will see.

![UI four screen](assets/four-screen.png)

> You can define code here for your test using [playwright.dev](https://playwright.dev). But please put your code specific between comments

Reviewer 2: ```suggestion
Here you can define the code for your test using [playwright.dev](https://playwright.dev). Put your code between the comments.
```

> 4. After set all values and define code of test what you want to run.

Reviewer 2: ```suggestion
4. After setting all the values and defining the test code,
```

>    You can you can click `Test locally` if everything works you will see metrics from one your test.

Reviewer 2: ```suggestion
click `Test locally` to check if everything works ad the logs are generated.
```

>    For Deploy your lambda function you can click `Deploy to the cloud`. You will see a status of all steps what is running behind. if Everything is good you need to see

Reviewer 2: ```suggestion
   To deploy the Lambda function, click `Deploy to the cloud`. You will see status of all the steps running in the background. The following message appears when the checks are completed:
```
